Feature/add allure reporting

ggarunchik opened the pull request.

ggarunchik-itechart committed:
Added allure reporting
Added CI (circleCi)

ggarunchik-itechart committed:
Added allure reporting
Added CI (circleCi)

ggarunchik-itechart committed:
Retry analizer has been activated

ggarunchik-itechart committed:
Retry analizer has been activated

dzmitryrak commented on src/test/java/tests/ui/login/LoginPageTests.java:
```diff
@@ -1,15 +1,18 @@
     private final String USER_PASSWORD = propertyReader.getPropertyValueByKey("password");
 
-    @Test(description = "Verifying that user can login with existing credits")
+    @Test(retryAnalyzer = RetryAnalyzer.class, description = "Verifying that user can login with existing credits")
```
давай подключать retryAnalyzer через regression.xml

dzmitryrak commented on src/test/java/tests/ui/login/LoginPageTests.java:
```diff
@@ -1,15 +1,18 @@
+    @Test(retryAnalyzer = RetryAnalyzer.class, description = "Verifying that user can login with existing credits")
     public void verifyUserCanLoginWithExistingCredits() {
         loginPageSteps.login(USER_NAME,USER_PASSWORD);
+        AllureUtils.takeScreenshot();
```
скриншот не делать тут

dzmitryrak replied:
скриншот в testlistener

ggarunchik-itechart committed:
Code refactor due to comments

ggarunchik-itechart committed:
Code refactor due to comments
fixed merge conflicts
added regression.xml and moved retry analyzer to the file

ggarunchik committed:
Merge branch 'master' into pr/2

ggarunchik committed:
fix merge conflict

ggarunchik commented:
fixed merge conflicts 
fixed comments
